### ISSUE TITLE
Per-Device Key Pair and General Purpose Sync

### ISSUE THREAD

**Issue by clashman** [irungentoo/toxcore#1356]
Assume that a separate key pair is used for each device of a user. This way, communication would not only happen between two users, but between two devices of one user as well. There would be the need to synchronise chatlogs and friendlists between devices. But beside syncing chatlogs, this functionality could also be used to sync arbitrary stuff, like files or entire folders. Not only on demand, but also continuously, if requested.

The following communication use cases are interesting:
* send a file to yourself (e.g. from laptop to mobile phone)
* use the messaging feature as a clipboard to send strings between your devices
* sync files between your devices or share them with others continously 

Additionally, with this prerequisite of a key pair for each device, the client discovery (pubkey -> IP) becomes a tremendously useful feature in itself: Say you want to ssh into one of your devices. You would be able to simply look up the corresponding IP in the DHT (if no firewall blocks access, of course).

**Comment by zetok** [comment]
>Say you want to ssh into one of your devices. You would be able to simply look up the corresponding IP in the DHT (if no firewall blocks access, of course).

Err, no. Tox DHT doesn't work like that.

**Comment by clashman** [comment]
@zetok The function DHT_addfriend looks to me as it does exactly that (Pubkey -> IP).
https://github.com/irungentoo/toxcore/blob/b8d530c9e0bbce331f22cbff7f818e49673d5bd7/toxcore/DHT.h#L269-270

**Comment by aaannndddyyy** [comment]
@clashman: yes those are very nice features, there have been several proposals for multi device support. With that in place, all this will be possible.
There's also a tox VPN in the works.

**Comment by Foxcool** [comment]
I came here to offer the same idea: a single user - a lot of devices. Each device has its own key, which avoids errors. The devices are synchronized with each other: logs, chats, files, etc. The device can be added to the synchronization and removed from it. It will be necessary to change the nature of the contact list: each user will have a lot of addresses, and write a message you need to be on any online address. Destination device are synchronized with each other.

But I see that a similar or more thoughtful already offered (:
